src/cloud-api-adaptor/pkg/securecomms/sshproxy/sshproxy.go

```diff
@@ -298,9 +299,15 @@ func (peer *SshPeer) Wait() {
 }
 
 func (peer *SshPeer) Close(who string) {
+	peer.mutex.Lock()
+	defer peer.mutex.Unlock()
+
 	if peer.terminated == "" {
 		logger.Printf("%s phase: peer done by >>> %s <<<", peer.phase, who)
 		peer.terminated = who
+		if peer.terminated == "" {
+			peer.terminated = "(unspecified)"
+		}
```

Comment:
Can we use [`sync.Once`](https://pkg.go.dev/sync#Once) here to simplify the logic?

Reply:
The function may run multiple times (for different peer objects). So it seems based on what I read that sync.Once does not fit here.

Reply:
I think you can add a `sync.Once` object into the `SshPeer` struct instead of `sync.Mutex`, then we can call `Close()` multiple times for different peer objects.

```
func (peer *SshPeer) Close(who string) {
    peer.once.Do(func() {
        logger.Printf("%s phase: peer done by >>> %s <<<", peer.phase, who)
        peer.sshConn.Close()
        close(peer.done)
   })
}
```

I think we can also eliminate the `terminate` field from the struct.